src/components/WeatherApp.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useCallback, useEffect } from "react";
+import styles from "../styles/WeatherApp.module.css";
+import getForecast from "../helpers/getForecast";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import SearchForm from "./SearchForm";
+import ErrorMessage from "./ErrorMessage";
+
+export default function WeatherApp() {
+  const [city, setCity] = useState("");
+  const [location, setLocation] = useState({});
+  const [errorMessage, setErrorMessage] = useState("");
+  const [forecasts, setForecasts] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(0);
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  const acknowledgeError = useCallback(() => {
+    setErrorMessage("");
+    setCity("");
+  }, []);
+
+  const handleForecastSelect = useCallback((date) => {
+    setSelectedDate(date);
+  }, []);
+
+  const updateCity = useCallback((event) => {
+    setCity(event.target.value);
+  }, []);
+
+  const submitCitySearch = useCallback(() => {
+    getForecast({
+      setLocation,
+      setForecasts,
+      setSelectedDate,
+      setErrorMessage,
+      city,
+    });
+  }, [city]);
+
+  useEffect(() => {
+    getForecast({
+      setLocation,
+      setForecasts,
+      setSelectedDate,
+      setErrorMessage,
+      city,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div className={`${styles["weather-app"]} shadow`}>
+      <SearchForm
+        handleChange={updateCity}
+        handleSubmit={submitCitySearch}
+        searchText={city}
+        placeholder="Enter city name"
+      />
+      {errorMessage ? (
+        <ErrorMessage
+          errorMessage={errorMessage}
+          handleClick={acknowledgeError}
+        />
+      ) : null}
```

Comment:
Rather than use a ternary here (and for the next 2 blocks) when you are not returning anything if its false, you could use the logical AND (&&) here instead:
```
      {errorMessage && (
        <ErrorMessage
          errorMessage={errorMessage}
          handleClick={acknowledgeError}
        />
      )}
```

Reply:
Hi Stu, I did have that originally, but for some reason, the falsey conditions were rendering the digit "0" in the browser.

Reply:
Hi Stu, I did have that originally, but for some reason, the falsey conditions were rendering the digit "0" in the browser.

Reply:
Ah, I think I see the problem here. That 0 is only returned from the two statements below this one (which works fine if using && operator). If the first condition is falsy, react wont return the second argument, returning instead the falsy value of the first argument (hence zero). If you cast these values to bools then itll work fine, eg:
```
      {!errorMessage && !!Object.keys(location).length && (
        <LocationDetails {...location} errorMessage={errorMessage} />
      )}
```